### PR TITLE
Add specs for paginator module. 

### PR DESCRIPTION
**Pull request by joshuatobin**
CC: @heroku/sre

### CODE REVIEW

File: app/support/mediators/incident/multi_syncher.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@ class MultiSyncher < Mediators::Base
     include Helpers::Paginator
 
     def call
-      fetch(Config.source_url).each do |incident|
+      Helpers::Paginator.fetch(Config.source_url).each do |incident|
```
Comment (jmervine) [inline app/support/mediators/incident/multi_syncher.rb]: You shouldn't need both the include and the full namespace in the call.

File: spec/lib/helpers/paginator_spec.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+describe Helpers::Paginator do
+
+  let(:url) { "https://example.localhost.com" }
+
+  def encoded_data
+    MultiJson.encode(incident_details)
+  end
+
+  def incident_details
+    {
```
Comment (jmervine) [inline spec/lib/helpers/paginator_spec.rb]: Totally up to you, but if you're json encoding these you can probably use the `incident_id: 900` format.

File: spec/lib/helpers/paginator_spec.rb [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      "resolved_at"=>"2016-06-02T12:26:33.061Z",
+      "review"=>true,
+    }
+  end
+
+  describe "#fetch" do
+    describe "with one request" do
+      before do
+        stub_request(:get, "#{url}/?page=1&per_page=100").to_return(
+        body: encoded_data,
```
Comment (jmervine) [inline spec/lib/helpers/paginator_spec.rb]: Tab here

File: spec/lib/helpers/paginator_spec.rb [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        headers: {"Link" => "<#{url}?page=1&per_page=100>; rel=\"last\", <#{url}?page=1&per_page=100>; rel=\"next\""}
```
Comment (jmervine) [inline spec/lib/helpers/paginator_spec.rb]: And here

File: spec/lib/helpers/paginator_spec.rb [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        Helpers::Paginator.fetch(url)
+
+        expect(WebMock).to have_requested(:get, "#{url}/?page=1&per_page=100").once
+      end
+    end
+
+    describe "Multiple requests" do
+      before do
+        stub_request(:get, /.*example.localhost.com.*/).to_return(
+        body: encoded_data,
```
Comment (jmervine) [inline spec/lib/helpers/paginator_spec.rb]: Tab here

File: spec/lib/helpers/paginator_spec.rb [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        headers: {"Link" => "<#{url}?page=2&per_page=100>; rel=\"last\", <#{url}?page=2&per_page=100>; rel=\"next\""}
```
Comment (jmervine) [inline spec/lib/helpers/paginator_spec.rb]: And here

File: spec/support/mediators/incident/multi_syncher_spec.rb [hunk 7]
```diff
@@ -32,11 +37,14 @@ def incident_details
   describe "#call" do
 
     before do
-      stub_request(:get, "https://status.localhost.com/*")
+        stub_request(:get, "#{url}/?page=1&per_page=100").to_return(
+        body: encoded_data,
+        headers: {"Link" => "<#{url}?page=1&per_page=100>; rel=\"last\", <#{url}?page=1&per_page=100>; rel=\"next\""}
+        )
```
Comment (jmervine) [inline spec/support/mediators/incident/multi_syncher_spec.rb]: I would drop this back a tab, or put it on the end of the previous line.

**Comment by jmervine** [comment]
A few stylistic suggestions, but this look great!

**Comment by joshuatobin** [comment]
@jmervine Addressed all comments.

**Comment by lexelby** [comment]
:+1:

File: spec/lib/helpers/paginator_spec.rb (hunk 2 above)
Comment (reidmix) [inline spec/lib/helpers/paginator_spec.rb]: Agreed, I prefer that format over the hashrockets.

File: spec/lib/helpers/paginator_spec.rb [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        )
+      end
+
+      it 'fetchs with multiple requests' do
+        Helpers::Paginator.fetch(url)
+
+        expect(WebMock).to have_requested(:get, /.*example.localhost.com.*/).twice
+
+        expect(WebMock).to have_requested(:get, "#{url}/?page=1&per_page=100").once
+        expect(WebMock).to have_requested(:get, "#{url}/?page=2&per_page=100").once
```
Comment (reidmix) [inline spec/lib/helpers/paginator_spec.rb]: Nice!

**Comment by reidmix** [comment]
👍

**Pull request merged**